evap/results/tools.py

```diff
@@ -393,23 +402,24 @@ def get_grade_color(grade):
 
 def textanswers_visible_to(contribution):
     if contribution.is_general:
-        contributors = (
-            UserProfile.objects.filter(
-                Q(
-                    contributions__evaluation=contribution.evaluation,
-                    contributions__textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
-                )
-                | Q(courses_responsible_for__in=[contribution.evaluation.course])
-            )
-            .distinct()
-            .order_by("last_name", "first_name")
-        )
+        contributors = set()
+        for other_contribution in contribution.evaluation.contributions.all():
+            if other_contribution.textanswer_visibility == Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS:
+                contributors.add(other_contribution.contributor)
+
+        contributors.update(contribution.evaluation.course.responsibles.all())
+        contributors = sorted(contributors, key=lambda user: (user.last_name, user.first_name))
     else:
         contributors = [contribution.contributor]
     non_proxy_contributors = [contributor for contributor in contributors if not contributor.is_proxy_user]
-    num_delegates = len(
-        set(UserProfile.objects.filter(represented_users__in=non_proxy_contributors).distinct()) - set(contributors)
+
+    num_delegates = sum(
+        1
+        for contributor in non_proxy_contributors
+        for delegate in contributor.delegates.all()
+        if delegate not in contributors
     )
```

Comment:
This is actually wrong, two contributors sharing a delegate would make that count go up by 2. I should add a test for that.